cnd/src/http_api/routes/rfc003/handlers/post_swap.rs

```diff
@@ -61,6 +110,87 @@ pub fn handle_post_swap<C: Connect>(
     }
 }
 
+fn initiate_request<D, AL, BL, AA, BA>(
+    dependencies: D,
+    swap_request: Request<AL, BL, AA, BA>,
+    bob_dial_info: DialInformation,
+    seed: Seed,
+) -> Result<(), metadata_store::Error>
+where
+    D: MetadataStore + StateStore,
+    AL: Ledger,
+    BL: Ledger,
+    AA: Asset,
+    BA: Asset,
+{
+    let counterparty = bob_dial_info.peer_id.clone();
+    let id = swap_request.id;
+
+    let metadata = Metadata::new(
+        id,
+        swap_request.alpha_ledger.into(),
+        swap_request.beta_ledger.into(),
+        swap_request.alpha_asset.into(),
+        swap_request.beta_asset.into(),
+        Role::Alice,
+        counterparty,
+    );
+
+    MetadataStore::insert(&dependencies, metadata)?;
+
+    let state = alice::State::proposed(swap_request.clone(), seed);
+    StateStore::insert(&dependencies, id, state);
+
+    Ok(())
+}
+
+fn send_request_and_spawn_alice<D, AL, BL, AA, BA>(
+    dependencies: D,
+    swap_request: Request<AL, BL, AA, BA>,
+    bob_dial_info: DialInformation,
+    seed: Seed,
+) where
+    LedgerConnectors: CreateLedgerEvents<AL, AA> + CreateLedgerEvents<BL, BA>,
+    D: MetadataStore + StateStore + SendRequest + Spawn,
+    AL: Ledger,
+    BL: Ledger,
+    AA: Asset,
+    BA: Asset,
+{
+    let future = {
+        async move {
+            let response = dependencies
+                .send_request(bob_dial_info.clone(), swap_request.clone())
+                .compat()
+                .await
+                .map_err(|e| {
+                    log::error!(
+                        "Failed to send swap request to {} because {:?}",
+                        bob_dial_info.clone(),
+                        e
+                    );
+                })?;
+
+            let id = swap_request.id;
+            let alice_state = match response.clone() {
+                Ok(accept) => State::accepted(swap_request.clone(), accept, seed),
+                Err(decline) => State::declined(swap_request.clone(), decline, seed),
+            };
+            StateStore::insert(&dependencies, swap_request.id, alice_state.clone());
+
+            let receiver = Spawn::spawn(&dependencies, swap_request, response);
+
+            tokio::spawn(receiver.for_each(move |update| {
```

Comment:
To harmonize this with the other usage, we could do this also outside of this function :)

Reply:
Good idea, will attempt.

Reply:
Discussed offline.